### DISCUSSION TITLE
Using zod to serialize data and remove data not present in the scheme

### DISCUSSION THREAD

**Discussion by derolf** [colinhacks/zod#743] [Q&A]
Let's assume I have a schema like:

```
const Person = z.object({name: z.string();});
```

Is there a way to use the `Person`-scheme to turn an object into a JSON while removing any additional values that are not present in the schema?

So, `Person.serialize({name: 'Foo', secret: 'Bar'})` should serialize to a JSON and remove `secret`

**Comment by scotttrinh** [comment]
By default, schemas will strip unknown keys, so this should work:

```ts
const json = JSON.stringify(Person.parse({ name: "Foo", secret: "Bar" }));
```

**Comment by JacobWeisenburger** [comment]
@scotttrinh suggestion works, or you can also use `.transform`

```ts
const Person = z.object( { name: z.string() } )
const personToString = Person.transform( JSON.stringify )

personToString.parse( { name: 'Foo' } )                // => {"name":"Foo"}
personToString.parse( { name: 'Foo', secret: 'Bar' } ) // => {"name":"Foo"}
personToString.parse( { name: 42 } )                   // throws error
```

More Info:
- [Unrecognized keys](https://github.com/colinhacks/zod#unrecognized-keys)
- [.transform](https://github.com/colinhacks/zod#transform)
